rosbag2_cpp/src/rosbag2_cpp/message_definitions/local_message_definition_source.cpp

```diff
@@ -234,7 +252,9 @@ rosbag2_storage::MessageDefinition LocalMessageDefinitionSource::get_full_text(
   Format format = Format::UNKNOWN;
   int32_t max_recursion_depth = ROSBAG2_CPP_LOCAL_MESSAGE_DEFINITION_SOURCE_MAX_RECURSION_DEPTH;
 
-  if (root_type.find("/srv/") == std::string::npos) {  // Not a service
+  if (root_type.find("/srv/") == std::string::npos &&
+    root_type.find("/action/") == std::string::npos)
+  {  // Neither srv nor action
```

Comment:
maybe this comment can be more straight.

```suggestion
  {  // Only msg and idl files
```